packages/prettier-plugin-java/src/printers/expressions.ts

```diff
@@ -119,19 +111,18 @@ export class ExpressionsPrettierVisitor extends BaseCstPrettierPrinter {
 
   lambdaParametersWithBraces(
     ctx: LambdaParametersWithBracesCtx,
-    params?: { shouldBreak?: boolean }
+    params?: { flat?: boolean }
   ) {
     const lambdaParameterList = this.visit(ctx.lambdaParameterList, params);
 
     if (findDeepElementInPartsArray(lambdaParameterList, ",")) {
-      const separator = params?.shouldBreak === false ? "" : softline;
-      const content = putIntoBraces(
```

Comment:
I think that there is some logic handling comments around braces in the putIntoBraces function: does it still works well if you remove the putIntoBraces method ?

Reply:
You're very right! This ended up sending me down quite a bit of a rabbit hole.

I realized that, because of the way `putIntoBraces` removes comments and puts them directly into the `Doc` it returns (rather than moving them to another token/node), it meant that certain code paths in which I leveraged the `conditionalGroup` function could result in comments being removed from the code entirely (😱).

I believe that I have now fixed that problem, and have added a bunch of tests for comments around methods and lambdas to ensure I handled all those cases properly.